packages/customize-widgets/src/components/inserter/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, createPortal } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { __experimentalLibrary as Library } from '@wordpress/block-editor';
+import { Button } from '@wordpress/components';
+import { useInstanceId } from '@wordpress/compose';
+
+/**
+ * External dependencies
+ */
+import { Dialog, DialogBackdrop } from 'reakit/Dialog';
+
+function useTargetDOM() {
+	const target = useRef();
+	if ( ! target.current ) {
+		target.current = document.getElementById( 'widgets-left' );
+	}
+	return target.current;
+}
+
+function Inserter( { ...props } ) {
+	const target = useTargetDOM();
+	const inserterTitleId = useInstanceId(
+		useRef(),
+		'customize-widget-layout__inserter-panel-title'
+	);
+
+	return createPortal(
+		<DialogBackdrop
+			className="customize-widgets-layout__inserter-panel-backdrop"
+			{ ...props }
+		>
+			<Dialog
+				{ ...props }
+				// Unset the "dialog" role to make it possible to be close the quick inserter on focus.
+				// See https://github.com/WordPress/gutenberg/pull/16964#discussion_r312015746 for more info.
+				role={ undefined }
+				className="customize-widgets-layout__inserter-panel"
+				hideOnClickOutside={ false }
+				aria-labelledby={ inserterTitleId }
+			>
+				<div className="customize-widgets-layout__inserter-panel-header">
+					<h2
+						id={ inserterTitleId }
+						className="customize-widgets-layout__inserter-panel-header-title"
+					>
+						{ __( 'Add a widget' ) }
+					</h2>
+					<Button
+						className="customize-widgets-layout__inserter-panel-header-close-button"
+						isTertiary
+						onClick={ props.hide }
+					>
+						{ __( 'DONE' ) }
```

Comment:
Why use _DONE_ here instead of an X icon? cc. @shaunandrews 

Also, I think better to use `__( 'Done' )` and `text-transform: uppercase` so that if we ever want to change the casing we don't have to re-translate the string.

Reply:
Ahh good point to the second part!

Reply:
I went with the "Done" button in hopes of avoiding confusion with the existing "X" icon, which exists the Customizer.

The inserter, when used in other places, doesn't have an explicit done button, so this is new UI. The block editor's inspector sidebar and modals use the "X" icon. The media and themes screen also use an "X" for their modals.

So we should probably try to match that pattern here.

<img width="769" alt="image" src="https://user-images.githubusercontent.com/191598/111187091-0b824580-858a-11eb-8a55-0b558bdd6b58.png">

I'm not in love with the dissonance between the close and search icons, so I added a border to help separate them. It helps a bit.